notes-with-indexeddb/notes.js

```diff
@@ -0,0 +1,169 @@
+/*
+Feature detection
+ */
+// DON'T use 'var indexedDB = ...' if you're not in a function.
+window.indexedDB = window.indexedDB || window.mozIndexedDB || window.webkitIndexedDB || window.msIndexedDB;
+
+// This line should only be needed if it is needed to support the object's constants for older browsers
+window.IDBTransaction = window.IDBTransaction || window.webkitIDBTransaction || window.msIDBTransaction || {
+    READ_WRITE: 'readwrite'
+};
+
+window.IDBKeyRange = window.IDBKeyRange || window.webkitIDBKeyRange || window.msIDBKeyRange;
+
+if (!window.indexedDB) {
+    window.alert("Your browser doesn't support a stable version of IndexedDB.");
+}
+
+/*
+showErrorMessage from event if available
+ */
+function showErrorMessage(event) {
+    if (event && event.target && event.target.errorCode) {
+        alert('Database error: ' + event.target.errorCode);
+        return;
+    }
+
+    alert('IndexedDB request denied');
+}
+
+/*
+addItem to db
+ */
+function addItem(db, data, success) {
+    var store = db.transaction(['notes'], 'readwrite').objectStore('notes'),
+        request = store.add(data);
+
+    request.onerror = showErrorMessage;
+    request.onsuccess = success;
+}
+
+function createIndex(event) {
+    var db = event.target.result,
+        store = db.createObjectStore('notes', {
+            autoIncrement: true
+        });
+
+    db.onerror = showErrorMessage;
+
+    // create indices to search db by
+    store.createIndex('title', 'title', {
+        unique: false
+    });
+    store.createIndex('description', 'description', {
+        unique: false
+    });
+}
+
+/*
+populateDb with data
+ */
+function populateDb(db, data) {
+    data.forEach(function(row) {
+        addItem(db, row);
+    });
+}
+
+/*
+removeItem from db
+ */
+function removeItem(db, key) {
+    var request = db.transaction(['notes'], 'readwrite')
+        .objectStore('notes')
+        .delete(key);
+
+    request.onsuccess = function(event) {
+        showNotes(db);
+    };
+}
+
+function showNotes(db) {
+    var list = document.createDocumentFragment(),
+        notes = document.getElementById('notes');
+
+    /*
+    listNotes with cursor
+     */
+    function listNotes(event) {
+        var cursor = event.target.result;
+
+        if (cursor) {
+            var title = document.createElement('h3'),
+                description = document.createElement('p'),
+                item = document.createElement('div'),
+                button = document.createElement('button');
+
+            title.innerHTML = cursor.value.title;
+            description.innerHTML = cursor.value.description;
```

Comment:
Please assign to `textContent` rather than `innerHTML` to avoid things accidentally being interpreted as markup.